### PR TITLE
adding expr to string for IsNotNull IsTrue IsFalse and IsUnkown

### LINKED ISSUE #9731: adding Expr->String for IsTrue, IsFalse, IsUnknown
### Is your feature request related to a problem or challenge?

_No response_

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Lordworms**
## Which issue does this PR close?

Closes #9731

### CODE REVIEW

File: datafusion/sql/src/unparser/expr.rs [hunk 1]
```diff
@@ -599,6 +611,19 @@ mod tests {
                 }),
                 "COUNT(DISTINCT *)",
             ),
+            (Expr::IsNotNull(Box::new(col("a"))), "\"a\" IS NOT NULL"),
+            (
+                Expr::IsTrue(Box::new((col("a") + col("b")).gt(lit(4)))),
+                "((\"a\" + \"b\") > 4) IS TRUE",
```
Comment (yyy1000) [inline datafusion/sql/src/unparser/expr.rs:617]: What about using raw strings? https://doc.rust-lang.org/reference/tokens.html#raw-string-literals
It matches the other tests. :)

Comment (Lordworms) [reply datafusion/sql/src/unparser/expr.rs:617]: Sure

**Review by alamb — APPROVED** [review]
This code looks good to me -- thank you @Lordworms 

I think it would be great to implement @yyy1000 's suggestion https://github.com/apache/arrow-datafusion/pull/9739/files#r1535890652 and then we can merge it.

cc @devinjdangelo

File: datafusion/sql/src/unparser/expr.rs [hunk 2]
```diff
@@ -599,6 +611,19 @@ mod tests {
                 }),
                 "COUNT(DISTINCT *)",
             ),
+            (Expr::IsNotNull(Box::new(col("a"))), r#""a" IS NOT NULL"#),
+            (
+                Expr::IsTrue(Box::new((col("a") + col("b")).gt(lit(4)))),
+                r#"(("a" + "b") > 4) IS TRUE"#,
```
Comment (alamb) [inline datafusion/sql/src/unparser/expr.rs:617]: 😍

**Review by alamb — APPROVED** [review]
Thanks again @Lordworms  and @yyy1000

**Pull request merged**